### DISCUSSION TITLE
Hardware requirements and config optimization

### DISCUSSION THREAD

**Discussion by kumbraGrr** [valhalla/valhalla#4774] [General]
Hi fellow Nordic mythology fans.

I was wondering if there is some benchmark results or recommendations for hardware requirements of the Valhalla?
If you have by hand any resources would be useful. What I need is not a world but one country area with high load.

Additionally, is there any utilization of the resources regarding number of threads to use and similar stuff that can be specified in configuration?
I saw  max_concurrent_reader_users property in config json but haven't found much of the documentation on it.

Thank you in advance,
Best

**Comment by kevinkreiser** [comment]
if you have more ram than the size of your data that would be the best case scenario. low latency disk is also useful to reduce the warm up time for the service. other than that you should be able to scale horizontally for each cpu you add so determining that would be something you can compute by estimating the amount of throughput you need and measuring the throughput of a single cpu.

I know it sounds generic but measuring is the best policy. in prod scenarios I've been responsible for I've used the m class aws instances with 8 cpus and 32gb of ram iirc this was for the whole planet as a dataset. if you have a dataset with historical traffic probably you want to double that
